extensions/data-transfer/portability-data-transfer-google/src/test/java/org/datatransferproject/datatransfer/google/media/GoogleMediaExporterTest.java

```diff
@@ -460,6 +472,101 @@ public void testGetGoogleMediaItemFailed() throws IOException, InvalidTokenExcep
     assertThat(end - start).isLessThan(RETRY_INTERVAL_MILLIS * RETRY_MAX_ATTEMPTS);
   }
 
+  @Test
+  public void testExportAlbums_failureInterruptsTransfer() throws Exception {
+    String ALBUM_ID_TO_FAIL_1 = "albumid3";
+    String ALBUM_ID_TO_FAIL_2 = "albumid5";
+
+    ImmutableList<PhotoModel> photos = ImmutableList.of();
+    ImmutableList<PhotoAlbum> albums = ImmutableList.of(
+        setUpSinglePhotoAlbum("albumid1", "album 1`", ""),
+        setUpSinglePhotoAlbum("albumid2", "album 2", ""),
+        setUpSinglePhotoAlbum(ALBUM_ID_TO_FAIL_1, "album 3", ""),
+        setUpSinglePhotoAlbum("albumid4", "album 4", ""),
+        setUpSinglePhotoAlbum(ALBUM_ID_TO_FAIL_2, "album 5", ""),
+        setUpSinglePhotoAlbum("albumid6", "album 6", "")
+    );
+
+    PhotosContainerResource container = new PhotosContainerResource(albums, photos);
+    ExportInformation exportInfo = new ExportInformation(null, container);
+
+    MediaMetadata photoMediaMetadata = new MediaMetadata();
+    photoMediaMetadata.setPhoto(new Photo());
+
+
+    // For the album_id_to_fail albums, throw an exception.
+    when(photosInterface.getAlbum(ALBUM_ID_TO_FAIL_1)).thenThrow(IOException.class);
+    when(photosInterface.getAlbum(ALBUM_ID_TO_FAIL_2)).thenThrow(IOException.class);
+    // For all other albums, return a GoogleMediaAlbum.
+    for (PhotoAlbum photoAlbum: albums) {
+      if (photoAlbum.getId().equals(ALBUM_ID_TO_FAIL_1) || photoAlbum.getId().equals(ALBUM_ID_TO_FAIL_2)) {
+        continue;
+      }
+      when(photosInterface.getAlbum(photoAlbum.getId())).thenReturn(
+          setUpGoogleAlbum(Optional.of(photoAlbum.getId()), Optional.of(photoAlbum.getName()))
+      );
+    }
+
+    assertThrows(IOException.class, () -> googleMediaExporter.export(
+        uuid, authData, Optional.of(exportInfo)
+    ));
+  }
+
+  @Test
+  public void testExportAlbums_retryingSkipsFailures() throws Exception {
+    String ALBUM_ID_TO_FAIL_1 = "albumid3";
+    String ALBUM_ID_TO_FAIL_2 = "albumid5";
+
+    ImmutableList<PhotoModel> photos = ImmutableList.of();
+    ImmutableList<PhotoAlbum> albums = ImmutableList.of(
```

Comment:
This is being repeated in both the tests. Let's move it up and make it static final too

Reply:
Won't move the photos and albums out of the test; this is the test data, and it shouldn't be obfuscated by being removed from the test